File: 2017.5.4/CurrencyRateObjC/CurrencyRateObjC/ViewController.m
//
//  ViewController.m
//  CurrencyRateObjC
//
//  Created by Benny Davidovitz on 04/05/2017.
//  Copyright © 2017 xcoder.solutions. All rights reserved.
//

#import "ViewController.h"
#import "Currency.h"
#import "AppManager.h"
#import "CurrencyCell.h"

@interface ViewController () <UICollectionViewDataSource, UICollectionViewDelegateFlowLayout>

@property (nonatomic, strong) NSArray <Currency*> *collectionArray;
@property (weak, nonatomic) IBOutlet UICollectionView *collectionView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    __weak ViewController *weakSelf = self;
    //__block int num = 5;
    
    [[AppManager sharedInstance] getCurrnciesWithBlock:^(NSArray<Currency *> *arr, NSError *err) {
        //num = 6;
        weakSelf.collectionArray = arr;
        [weakSelf.collectionView reloadData];
    }];
    
    /*
    NSDictionary *testDictionary = @{
                                     @"currency_code":@"AED",
                                     @"rate": @3.6725,
                                     @"code": @"AE",
                                     @"name": @"United Arab Emirates"
                                     };
    
    Currency *currency = [[Currency alloc] initWithDictionary:testDictionary];
    */
}


#pragma mark - UICollectionView -
- (NSInteger) numberOfSectionsInCollectionView:(UICollectionView *)collectionView{
    return 1;
}

- (NSInteger) collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section{
    return [_collectionArray count];
}

- (__kindof UICollectionViewCell *) collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath{
    
    CurrencyCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"cell" forIndexPath:indexPath];
    
    Currency *obj = _collectionArray[indexPath.row];
    [cell configureWithCurrency:obj];
    
    return cell;
    
}

- (CGSize) collectionView:(UICollectionView *)collectionView layout:(UICollectionViewLayout *)collectionViewLayout sizeForItemAtIndexPath:(NSIndexPath *)indexPath{
    
    return collectionView.bounds.size;
    
}



@end

File: 2017.4.6/CollectionViewProject/CollectionViewProject/ViewController.swift
//
//  ViewController.swift
//  CollectionViewProject
//
//  Created by hackeru on 06/04/2017.
//  Copyright © 2017 hackeru. All rights reserved.
//

import UIKit

class ViewController: UIViewController , UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout{

    func numberOfSections(in collectionView: UICollectionView) -> Int {
        return 10
    }
    
    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return section + 1
    }
    
    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as! Cell
        
        cell.label.text = "Item \(indexPath.item)"
        
        return cell
    }
    
    func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
        //UICollectionElementKindSectionHeader
        let header = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: "header", for: indexPath) as! HeaderView
        
        header.label.text = "Section \(indexPath.section)"
        
        return header
    }
    
    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
        
        let s = min(collectionView.bounds.width, collectionView.bounds.height) / 3
        
        return CGSize(width: s, height: s)
        
    }

}













